### ISSUE TITLE
[Serve] Expose multiple ports while using sky serve

### ISSUE THREAD

**Issue by aswanthkrishna** [skypilot-org/skypilot#3727]
expose multiple ports while using sky serve

I would like the ability to add multiple ports in the resource definition in sky serve.  This might be especially useful when i want to deploy applications with a separate frontend and backed server.

**Comment by Jayasankar-kk** [comment]
Serving without mentioning ports,
We are unable to serve without mentioning ports under resources. I think this is also a similar issue

When I tried sky serve with lambda cloud I got the following error
"ValueError: No enabled clouds support opening ports. To fix: do not specify resources.ports, or enable a cloud that does support this feature."
When I tried serving without mentioning the ports in the yaml file, I got the following error
"ValueError: Must only specify one port in resources. Each replica will use the port specified as application ingress port."

**Comment by info-wordcab** [comment]
@cblmemo Following up to what Adrian mentioned in [this issue](https://github.com/skypilot-org/skypilot/issues/3621), this works if you comment out [this error](https://github.com/skypilot-org/skypilot/blob/e794971548584cdc9c8d6c1bde4b734c151ef3c4/sky/cli.py#L3918) in `cli.py`, and [this error](https://github.com/skypilot-org/skypilot/blob/e794971548584cdc9c8d6c1bde4b734c151ef3c4/sky/serve/core.py#L79) in `serve/core.py`, with the caveat that the lowest port is selected. So if your FastAPI server is on `5001`, and Grafana is on `3000`, you'd switch Grafana to `5005` or something higher.

**Comment by romilbhardwaj** [comment]
Being fixed in #4356.

**Comment by aleksandr-smechov** [comment]
@Conless one side issue here, is that if we're setting up something like Triton, we'd need separate ports for health check (HTTP) and the LB (gRPC).

**Comment by cblmemo** [comment]
Closed due to #4356 being merged. @aleksandr-smechov , could you help confirm if the fix works for you?

**Comment by aleksandr-smechov** [comment]
@cblmemo We just put an intermediary FastAPI server on the same instance, which routed things appropriately. You can only use one port with the LB endpoint, right? 

The alternative would be to find out the IP addresses available under that LB and use the different ports that way.

**Comment by cblmemo** [comment]
> You can only use one port with the LB endpoint, right?

Yes; for now we only support one LB per service. We'll consider multiple LB in the following development :)) Thanks for the feedback!

**Comment by cblmemo** [comment]
> [@cblmemo](https://github.com/cblmemo) We just put an intermediary FastAPI server on the same instance, which routed things appropriately. You can only use one port with the LB endpoint, right?
> 
> The alternative would be to find out the IP addresses available under that LB and use the different ports that way.

Keep track of this in #4869.
